core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -623,8 +623,9 @@ private[spark] class TaskSetManager(
    *
    * It is possible that this taskset has become impossible to schedule *anywhere* due to the
    * blacklist.  The most common scenario would be if there are fewer executors than
-   * spark.task.maxFailures. We need to detect this so we can fail the task set, otherwise the job
-   * will hang.
+   * spark.task.maxFailures. We need to detect this so we can avoid the job from being hung.
+   * If dynamic allocation is enabled we try to acquire new executor/s by killing the existing one.
+   * In case of static allocation we abort the taskSet immediately to fail the job.
```

Comment:
why do you want something different with static allocation?  If you kill an executor, static allocation will also request a replacement.

Reply:
Yes. The change of removing a single executor takes care of static allocation as well. I will update the comments.